### PR TITLE
support python test without installation python package

### PR DESCRIPTION
**Pull request by jacquesqiao**
fix https://github.com/PaddlePaddle/Paddle/issues/3352
after `make`, can use
```shell
ctest -R test_add
```
to run python unit test is build dir without install the python package.

tested on all python unit test.

### CODE REVIEW

File: cmake/generic.cmake [hunk 1]
```diff
@@ -411,7 +411,7 @@ function(py_test TARGET_NAME)
     set(multiValueArgs SRCS DEPS)
     cmake_parse_arguments(py_test "${options}" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})  
     add_test(NAME ${TARGET_NAME}
-             COMMAND env PYTHONPATH=${PADDLE_PYTHON_PACKAGE_DIR}
+             COMMAND env PYTHONPATH=${PADDLE_PYTHON_LIB_DIR}
```
Comment (gangliao) [inline cmake/generic.cmake]: Did you test swig test??

Comment (jacquesqiao) [inline cmake/generic.cmake]: yes, I have run all the python test and I don't think that copy all the generated files to python src dir is a good idea.

Comment (gangliao) [inline cmake/generic.cmake]: please `pip uninstall paddlepaddle` before you test this PR.
```shell
test 91
    Start 91: testGradientMachine

91: Test command: /usr/bin/env "PYTHONPATH=" "python2" "testGradientMachine.py"
91: Test timeout computed to be: 9.99988e+06
91: Traceback (most recent call last):
91:   File "testGradientMachine.py", line 15, in <module>
91:     from py_paddle import swig_paddle
91: ImportError: No module named py_paddle
1/1 Test #91: testGradientMachine ..............***Failed    0.03 sec

0% tests passed, 1 tests failed out of 1

test 90
    Start 90: testArguments

90: Test command: /usr/bin/env "PYTHONPATH=" "python2" "testArguments.py"
90: Test timeout computed to be: 9.99988e+06
90: Traceback (most recent call last):
90:   File "testArguments.py", line 15, in <module>
90:     from py_paddle import swig_paddle
90: ImportError: No module named py_paddle
1/1 Test #90: testArguments ....................***Failed    0.03 sec
```

Comment (gangliao) [inline cmake/generic.cmake]: `copy` Python proto is necessary. because we import proto like `import paddle.v2.framework.proto` in our code.

**Pull request merged**